.semaphore/prerelease-multi-arch-packaging.yml

```diff
@@ -0,0 +1,61 @@
+version: v1.0
+name: prerelease-multi-arch-packaging
+agent:
+  machine:
+    type: s1-prod-ubuntu24-04-amd64-1
+
+global_job_config:
+  prologue:
+    commands:
+      - checkout
+
+blocks:
+  - name: "Package VSIX Files"
+    dependencies: []
+    task:
+      prologue:
+        commands:
+          - git fetch --all
+          - git checkout $COMMIT_SHA
+          - . vault-setup
+          - make install-dependencies
+      jobs:
+        - name: "Package VSIX Files"
+          matrix:
+            - env_var: TARGET
+              values:
+                - darwin-x64
+                - darwin-arm64
+                - linux-x64
+                - linux-arm64
+                - win32-x64
+          commands:
+            - |
+              case "$TARGET" in
+                darwin-x64)  export SIDECAR_OS_ARCH=macos-amd64 ;;
+                darwin-arm64) export SIDECAR_OS_ARCH=macos-arm64 ;;
+                linux-x64)   export SIDECAR_OS_ARCH=linux-amd64 ;;
+                linux-arm64) export SIDECAR_OS_ARCH=linux-arm64 ;;
+                win32-x64)   export SIDECAR_OS_ARCH=windows-x64 ;;
+                *) echo "Unknown TARGET: $TARGET" && exit 1 ;;
+              esac
```

Comment:
We will likely want to migrate https://github.com/confluentinc/vscode/blob/main/.semaphore/multi-arch-packaging.yml to this setup as well since it removes the need for multiple agents (and powershell)